types/supports-color/index.d.ts

```diff
@@ -13,8 +14,24 @@ export namespace supportsColor {
         has16m: boolean;
     }
 
+    interface Options {
+        /**
+         * By default it is `true`, which instructs `supportsColor()` to sniff `process.argv`
+         * for the multitude of `--color` flags (see _Info_ below).
+         * If `false`, then `process.argv` is not considered when determining color support.
+         * @default true
+         */
+        sniffFlags?: boolean;
+        isTTY?: boolean;
+    }
+
     type SupportsColor = false | Level;
 }
 
+export function supportsColor(
+    stream: NodeJS.ReadStream | NodeJS.WriteStream,
```

Comment:
Does it really handle ReadStreams?  What does it mean for a ReadStream to support color?  https://github.com/chalk/supports-color/compare/v7.2.0...v8.1.1#diff-5a831ea67cf5cf8703b0de46901ab25bd191f56b320053be9332d9a3b0d01d15R41

Reply:
I've made it generic stream, IMO this could even be an object that has this simple `isTTY` bool property as a contract, so it would match anything, including package own internal tests (it has to be {} in this scenario).

Reply:
If this package isn't exclusively used with node, it might be nice to avoid pulling in the node types.

Reply:
i'll drop and fallback to contract (via interface), one sec

Reply:
settled on this little optional contract and object. It supports whatever, but this would really be counter-productive here, usually this is just an object with optional contract. Thx!